dashboard/app/controllers/api_controller.rb

```diff
@@ -546,6 +547,120 @@ def assessments_structure
     render json: assessments
   end
 
+  # Return a hash by student_id for all students in a section. The value is a hash by
+  # script_level_id, with information on the student responses for that assessment.
+  # Currently, very similar logic to section_assessments, but will change as we refactor. The format
+  # of the results is different enough that I'm duplicating some code it for now.
+  # TODO(caleybrock): currently only used in internal experiment, must add controller tests.
+  def assessments_responses
```

Comment:
Is `ApiController` really the right place for this? It seems like it's become a grab-bag of barely-related routes.  Maybe not necessary to fix in this PR, but worth considering a reorganization soon.

Reply:
I agree - probably worth me discussing this topic with someone. I've been adding new things here because their old versions were already here, but it feels messy.